src/coreclr/src/vm/pinvokeoverrideimpl.cpp

```diff
@@ -2,31 +2,32 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 //
 
-//
-// REVIEW: THIS IS A TEST-ONLY IMPLEMENTATION AND WILL BE REMOVED.
-// THE ACTUAL IMPLEMENTATION IS PROVIDED BY THE HOST
-//
-
 #include "common.h"
 #include "pinvokeoverrideimpl.h"
 
 extern "C" const void* GlobalizationResolveDllImport(const char* name);
 extern "C" const void* CompressionResolveDllImport(const char* name);
 
-const void* __stdcall SuperHost::ResolveDllImport(const char* libraryName, const char* entrypointName)
+const void* __stdcall DefaultPInvokeOverride::ResolveDllImport(const char* libraryName, const char* entrypointName)
```

Comment:
Does coreclr really need to have any hardcoded default overrides? Would it be better for all of it to be in the host?

Reply:
Probably. I've just realized, that we need to always override Globalization stuff, since it is always embedded. Either every host passes an override for it or we have some kind of default.

Reply:
I need a default override at least for now - to see what could be broken on platform combinations that I did not test locally.

Requiring hosts to always pass an override, at least for Globalization, would work too.

Reply:
I wonder if we could have any compat concerns if we start requiring a Globalization override from all hosts. 
In such case a default override could be a viable solution.

Reply:
I see - if we keep statically linking globalization into libcoreclr.so, it does not really make sense (nor it is even possible) for the hosts to supply it. I think we should either split globalization back into separate .dll; or introduce concept of 
"built-in override" that captures whatever is linked into libcoreclr.so and call it in addition to the host supplied one.

Reply:
Hosts that statically link coreclr with other stuff (including Globalization) may still be able to provide Globalization override. It only feels slightly unnatural, since they have no choice. 

An embedder that hosts .so dynamically can’t provide such override easily.

Yes, think we need a built-in default at least for the .so case.

Also, it maybe still possible that embedder can and wants to override that, so host’s override should run before the  built-in one. (not as either/or as in the last change).

Reply:
Agree. Sounds good.